packages/network-controller/CHANGELOG.md

```diff
@@ -7,6 +7,41 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
+- Add `removeNetwork`, which replaces `removeNetworkConfiguration` and can be used to remove existing network clients for a chain ([#4268](https://github.com/MetaMask/core/pull/4286))
+- Add `getDefaultNetworkControllerState` function, which replaces `defaultState` and matches patterns in other controllers ([#4268](https://github.com/MetaMask/core/pull/4286))
+- **BREAKING:** Replace `NetworkConfiguration` type with a new definition ([#4268](https://github.com/MetaMask/core/pull/4286))
+  - A network configuration no longer represents a single RPC endpoint, but a collection of RPC endpoints that can all be used to interface with a single chain.
+  - The only property that has been retained on this type is `chainId`.
+  - `ticker` has been renamed to `nativeTokenName`.
+  - `nickname` has been renamed to `name`.
+  - `blockExplorerUrl` has been pulled out of `rpcPrefs`.
+  - `defaultRpcEndpointUrl` has been added.
+  - `rpcEndpoints` has been added as well. This is an an array of objects, where each object has properties `name`, `networkClientId` (optional), `type`, and `url`.
+  - `id` has been removed. Previously, this represented the ID of the network client associated with the network configuration. Since network clients are now created from RPC endpoints, the equivalent to this is the `networkClientId` property on an `RpcEndpoint`.
+- **BREAKING:** The network controller messenger must now allow the action `NetworkController:getNetworkConfigurationByChainId` ([#4268](https://github.com/MetaMask/core/pull/4286))
+- **BREAKING:** The network controller messenger must now allow the event `NetworkController:networkAdded` ([#4268](https://github.com/MetaMask/core/pull/4286))
+- **BREAKING:** The `NetworkController` constructor will now throw if invalid initial state is provided ([#4268](https://github.com/MetaMask/core/pull/4286))
+  - `networkConfigurationsByChainId` cannot be empty.
+  - The `chainId` of a network configuration in `networkConfigurationsByChainId` must match the chain ID it is filed under.
+  - The `defaultRpcEndpointUrl` of a network configuration in `networkConfigurationsByChainId` must match an entry in its `rpcEndpoints`.
+  - `selectedNetworkClientId` must match the `networkClientId` of an RPC endpoint in `networkConfigurationsByChainId`.
+- **BREAKING:** Update `getNetworkConfigurationByNetworkClientId` so that when given an Infura network name (that is, a value from `InfuraNetworkType`), it will return a masked version of the RPC endpoint URL for the associated Infura network ([#4268](https://github.com/MetaMask/core/pull/4286))
+  - If you want the unmasked version, you'll need the `url` property from the network _client_ configuration, which you can get by calling `getNetworkClientById` and then accessing the `configuration` property off of the network client.
```

Comment:
As an alternative to requiring this step, I could make `getNetworkConfigurationByNetworkClientId` (or any other methods that return a single network configuration or multiple network configurations) unmask the Infura API URL automatically. Thoughts?